### PR TITLE
Added the conditional-installer module

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a pull request! Maybe @vpodzime can review this?

File: cfbs.json [hunk 1]
```diff
@@ -13,6 +13,55 @@
       "subdirectory": "reporting/client-initiated-reporting",
       "steps": ["json def.json def.json"]
     },
+    "conditional-installer": {
+      "description": "Allows you to specify a list of packages you want uninstalled on your hosts.",
```
Comment (vpodzime) [inline cfbs.json:17]: This is a weird description for something that is called *conditional-**installer***

File: cfbs.json [hunk 2]
```diff
@@ -13,6 +13,55 @@
...
+              "key": "name",
+              "type": "string",
+              "label": "Package name",
+              "question": "Package name"
+            },
+            {
+              "key": "condition",
+              "type": "string",
+              "label": "Condition",
+              "question": "Condition"
```
Comment (vpodzime) [inline cfbs.json]: Should mention this is a class (expression), I think.

**Review by vpodzime — COMMENTED** [review]
Might as well squash the commits into a single one. Looks good to me otherwise.

File: security/conditional-installer/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+`talk` and `samba` will be uninstalled everywhere.
+As always with CFEngine, if the state is already correct, if the packages are already installed / not installed, no actions will be performed.
+
+**Hint:** Package names are based on your system's package manager.
+In the list of packages to uninstall, it is convenient to just name all the variants of the name, for example `httpd` for RHEL and `apache2` for Debian-based systems.
+For the packages to install you have to be more careful; when package names are different use the platform as part of the condition and the correct package name for that platform.
+For our example above, it could be install `apache2` with the condition `webserver&debian`, and another entry to install `httpd` with the condition `webserver&redhat`.
+
+**Note:** This module is experimental, and things might change.
+Specifically the module input currently accepts strings with comma separated package names.
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply security/conditional-installer/README.md:19]: Yea, maybe a good idea. I kinda wanted to merge this in a somewhat "experimental" state and play around with how it looks (both on website, in MP and in terminal).

File: cfbs.json (hunk 1 above)
Comment (olehermanse) [reply cfbs.json]: Fixed, thanks.

File: security/conditional-installer/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (olehermanse) [reply security/conditional-installer/README.md:19]: Left for later.

File: cfbs.json (hunk 2 above)
Comment (olehermanse) [reply cfbs.json]: Left for later.

**Comment by olehermanse** [comment]
> Might as well squash the commits into a single one.

Done

File: cfbs.json [hunk 4]
```diff
@@ -13,6 +13,55 @@
...
+          "type": "list",
+          "variable": "packages_to_install",
+          "namespace": "conditional_installer",
+          "bundle": "conditional_installer",
+          "label": "Install",
+          "subtype": [
+            {
+              "key": "name",
+              "type": "string",
+              "label": "Package(s)",
```
Comment (nickanderson) [inline cfbs.json:38]: Package(s)? It's not clear here how I would enter multiple into a single string. Space vs comma separated? Perhaps update the label or question to provide that hint?

File: cfbs.json [hunk 5]
```diff
@@ -13,6 +13,55 @@
...
+              "question": "Package(s) to install"
+            },
+            {
+              "key": "condition",
+              "type": "string",
+              "label": "Condition",
+              "question": "Condition for where to install"
```
Comment (nickanderson) [inline cfbs.json:45]: As a CFEngine policy person, it's not clear if this should be a class singular or allows a class expression.

Comment (nickanderson) [reply cfbs.json:45]: Oh, it's in the readme :+1:

File: security/conditional-installer/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+This is an experimental module for both uninstalling and installing packages based on conditions (CFEngine class expressions).
+Users can specify a list of packages which should be uninstalled by default, and some packages which should be installed under certain conditions.
+With this logic, you can have a long list of packages which are generally not allowed, and then for some of them, specify the exact scenarios where they are allowed.
+This module uses the system package manager (via packages promises) to both install and uninstall packages.
```
Comment (nickanderson) [inline security/conditional-installer/README.md:4]:
```suggestion
This module uses the default system package manager (via package_module based packages promises) to both install and uninstall packages.
```

File: cfbs.json (hunk 4 above)
Comment (nickanderson) [reply cfbs.json:38]: Oh, I see it's in the readme. :+1:

**Review by nickanderson — APPROVED** [review]
Looks ok to me. Haven't tried it myself.

File: security/conditional-installer/main.cf [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+{
+  classes:
+    "enabled"
+      if => and(isvariable(packages_to_install), isvariable(packages_to_uninstall));
+
+  vars:
+    # Mock data, should be based on input:
+    # "packages_to_uninstall"
+    #   string => 'talk,talk-server,telnet,telnet-server';
+#
```
Comment (vpodzime) [inline security/conditional-installer/main.cf:16]: I see an outlier here.

**Review by vpodzime — APPROVED** [review]
Looks good to me otherwise.

File: security/conditional-installer/main.cf (hunk 7 above)
Comment (olehermanse) [inline security/conditional-installer/main.cf:16]:
```suggestion
    # 
```

**Pull request merged**